crates/noirc_evaluator/src/ssa/anchor.rs

```diff
@@ -0,0 +1,263 @@
+                    if by_type.contains_key(&res_type) {
+                        let tu = by_type[&res_type];
+                        if let Some(ins) = igen.try_get_instruction(tu) {
+                            if !ins.is_deleted() {
+                                return Some(tu);
+                            }
+                        }
+                    }
+                }
+            }
+            _ => unreachable!(),
+        }
+
+        None
+    }
+
+    fn get_mem_map(&self, a: ArrayId) -> &Vec<VecDeque<(usize, NodeId)>> {
+        &self.mem_map[&a]
+    }
+
+    pub fn get_mem_all(&self, a: ArrayId) -> &VecDeque<MemItem> {
+        &self.mem_list[&a]
+    }
+
+    pub fn use_array(&mut self, a: ArrayId, len: usize) {
+        if !self.mem_list.contains_key(&a) {
+            let def: VecDeque<(usize, NodeId)> = VecDeque::new();
+            self.mem_list.entry(a).or_insert_with(VecDeque::new);
+            self.mem_map.entry(a).or_insert_with(|| vec![def; len]);
+        }
+    }
+
+    pub fn push_mem_instruction(&mut self, ctx: &SsaContext, id: NodeId) {
+        let ins = ctx.get_instruction(id);
+        let (array_id, index, is_load) = Anchor::get_mem_op(&ins.operation);
+        self.use_array(array_id, ctx.mem[array_id].len as usize);
+        let prev_list = self.mem_list.get_mut(&array_id).unwrap();
+        let len = prev_list.len();
+        if let Some(index_value) = ctx.get_as_constant(index) {
+            let mem_idx = index_value.to_u128() as usize;
+            let last_item = prev_list.front();
+            let mut new_item = None;
+            match last_item {
+                Some(MemItem::Const(pos)) => {
+                    self.mem_map.get_mut(&array_id).unwrap()[mem_idx].push_front((*pos, id))
+                }
+                Some(MemItem::ConstLoad(pos)) => {
+                    if is_load {
+                        self.mem_map.get_mut(&array_id).unwrap()[mem_idx].push_front((*pos, id));
+                    } else {
+                        new_item = Some(MemItem::Const(pos + 1));
+                        self.mem_map.get_mut(&array_id).unwrap()[mem_idx].push_front((pos + 1, id));
+                    }
+                }
+                None | Some(MemItem::NonConst(_)) => {
+                    if is_load {
+                        prev_list.push_front(MemItem::ConstLoad(len));
+                        self.mem_map.get_mut(&array_id).unwrap()[mem_idx].push_front((len, id));
+                    } else {
+                        prev_list.push_front(MemItem::Const(len));
+                        self.mem_map.get_mut(&array_id).unwrap()[mem_idx].push_front((len, id));
+                    }
+                }
+            }
+            if let Some(item) = new_item {
+                prev_list.push_front(item);
+            }
+        } else {
+            prev_list.push_front(MemItem::NonConst(id));
+        }
+    }
+
+    pub fn find_similar_mem_instruction2(
```

Comment:
Lets rename this, I'm unsure what it is expected to do differently from the name

Reply:
Indeed it was a temp name that I forgot to change. I removed the "2".